OSSM-11032 [DOCS] Modify upgrade procedure for ambient mode (#101283)
openshift/openshift-docs: modules/ossm-verifying-l7-features-with-authorization-policies.adoc

// Module included in the following assemblies:
// update/ossm-updating-openshift-service-mesh-in-ambient-mode.adoc

:_mod-docs-content-type: PROCEDURE
[id="ossm-verifying-l7-features-with-authorization-policies_{context}"]
= Verifying Layer 7 (L7) features with authorization policies

[role="_abstract"]

> After updating the waypoint proxies, verify that Layer 7 (L7) authorization policies are enforced correctly by the waypoint proxy. In this verification, you test the `productpage-waypoint` `AuthorizationPolicy` from the Security Authorization section, which allows only the `curl` service with the `default/curl` service account to send `GET` requests to the `productpage` service.

Reviewer: ```suggestion
After updating the waypoint proxies, if in your cluster you are using Layer 7 (L7) authorization policies, you need to verify that they are enforced correctly by the new waypoint proxy. In this verification example, we are using the `productpage-waypoint` `AuthorizationPolicy` from the Security Authorization section, which allows only the `curl` service with the `default/curl` service account to send `GET` requests to the `productpage` service.
```

> After updating the waypoint proxies, if in your cluster you are using Layer 7 (L7) authorization policies, you need to verify that they are enforced correctly by the new waypoint proxy. In this verification example, we are using the `productpage-waypoint` `AuthorizationPolicy` from the Security Authorization section, which allows only the `curl` service with the `default/curl` service account to send `GET` requests to the `productpage` service.

Reviewer: Closing this and adding the suggested change to add the proper example from upstream documentation

.Prerequisites

> * You have updated the waypoint proxies.

Reviewer: ```suggestion
After updating the waypoint proxies, if in your cluster you are using Layer 7 (L7) authorization policies, you need to verify that they are enforced correctly by the new waypoint proxy. In this verification example, we are using an `AuthorizationPolicy` with the name `productpage-waypoint`, which allows only requests with the `default/sa/curl` service account to send `GET` requests to the `productpage` service.

.Prerequisites

* You have updated the waypoint proxies.
* You have created an application pod using the described service account in the `AuthorizationPolicy`.
* You have created an `AuthorizationPolicy`. For this example, you can create the resource by running:

[source,terminal]
----
oc apply -f - <<EOF
apiVersion: security.istio.io/v1
kind: AuthorizationPolicy
metadata:
  name: productpage-waypoint
  namespace: bookinfo
spec:
  targetRefs:
  - kind: Service
    group: ""
    name: productpage
  action: ALLOW
  rules:
  - from:
    - source:
        principals:
        - cluster.local/ns/default/sa/curl
    to:
    - operation:
        methods: ["GET"]
EOF
----
```

.Procedure

> . Verify that services not included in the allow list, such as the ratings service, are denied access, by running the following command:

Reviewer 2: Did we miss including the AuthorizationPolicy?

= Verifying Layer 7 (L7) features with authorization policies

[role="_abstract"]
After updating the waypoint proxies, verify that the Layer 7 (L7) authorization policies are enforced correctly. In this example, the `AuthorizationPolicy` resource named `productpage-waypoint` allows only requests from the `default/sa/curl` service account to send `GET` requests to the `productpage` service.

.Prerequisites

* You have updated the waypoint proxies.

* You have created an application pod using the described service account in the `AuthorizationPolicy` resource.

* You have created an `AuthorizationPolicy` resource.

.Procedure

. Optional: Create the `AuthorizationPolicy` resource if it does not already exist by running the following command:
+
[source,terminal]
----
$ oc apply -f - <<EOF
apiVersion: security.istio.io/v1
kind: AuthorizationPolicy
metadata:
  name: productpage-waypoint
  namespace: bookinfo
spec:
  targetRefs:
  - kind: Service
    group: ""
    name: productpage
  action: ALLOW
  rules:
  - from:
    - source:
        principals:
        - cluster.local/ns/default/sa/curl
    to:
    - operation:
        methods: ["GET"]
EOF
----

> . Verify that services not included in the allow list, such as the ratings service, are denied access, by running the following command:

Reviewer 3: ```suggestion
. Verify that services not included in the allow list, such as the ratings service, are denied access by running the following command:
```
